feat: add lint package

mfrancis-ccc opened the pull request.

jwatson-ccc commented:
We will also need to update the Repo to make this a separate package that is released since this is a mono-repo.

This is in the release-please configuration

mfrancis-ccc commented:
I refactored to be a single lint submodule with subpackages for each analyzer. I also added an analyzer for error wrapping, added a readme, and added the release-please config for a new submodule.

I still need to test the readme instructions, but I don't think it can be tested until it's in master as the `.golangci.yml` doesn't have an option to set the branch. I based the instructions on what their docs say, so hopefully it works as is.

jwatson-ccc reviewed (approved):
LGTM
